### DISCUSSION TITLE
Mapboxgl load Martin generated vector slices, small-scale automatic simplification, how to solve?

### DISCUSSION THREAD

**Discussion by szygiser** [maplibre/martin#1077] [Q&A]
![b6cc18b4cf88b6ad7af92e2f3a8f792](https://github.com/maplibre/martin/assets/15412933/35826fed-70df-4a80-b1a1-096546c797f6)

Mapboxgl load Martin generated vector slices, small-scale automatic simplification, how to solve?

**Comment by nyurik** [comment]
@szygiser I do not understand the question. Please provide more details

**Comment by szygiser** [comment]
> @szygiser I do not understand the question. Please provide more details

I used mpboxgl to load the vector tiles of the line layer generated by Martin. When I reduced the map scale, the line layer became blurred。

**Comment by szygiser** [comment]
> @szygiser I do not understand the question. Please provide more details

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Add a third party vector tile source</title>
    <meta name="viewport" content="initial-scale=1,maximum-scale=1,user-scalable=no">
    <link href="https://api.mapbox.com/mapbox-gl-js/v2.14.1/mapbox-gl.css" rel="stylesheet">
    <script src="https://api.mapbox.com/mapbox-gl-js/v2.14.1/mapbox-gl.js"></script>
    <style>
        body {
            margin: 0;
            padding: 0;
        }

        #map {
            position: absolute;
            top: 0;
            bottom: 0;
            width: 100%;
        }
    </style>
</head>
<body>
<div id="map"></div>
<script>
    // TO MAKE THE MAP APPEAR YOU MUST
    // ADD YOUR ACCESS TOKEN FROM
    // https://account.mapbox.com
    mapboxgl.accessToken = 'pk.eyJ1Ijoic29uZ3plbmd5dSIsImEiOiJjbGo0ZDM0bmMwMTBzM2RreGlqMXNsc2YwIn0.ik-fY3Yx7mRu02ySZnjavQ';
    const map = new mapboxgl.Map({
        container: 'map',
        // Choose from Mapbox's core styles, or make your own style with Mapbox Studio
        style: 'mapbox://styles/mapbox/light-v11',
        zoom: 10,
        center: [117.110805, 36.663046],
        projection: 'mercator',
        antialiased: true,
        hash: true,
        minTileCacheSize: 512,
        collectResourceTiming: true,
        cooperativeGestures: false,
        crossSourceCollisions: true,
        refreshExpiredTiles: true,
        trackResize: false
    });

    map.on('load', () => {
            addMvt();
        }
    );

    map.addControl(new mapboxgl.NavigationControl());

    function addMvt() {
        map.addLayer({
            id: 'lines',
            type: 'line',
            source: {
                type: 'vector',
                url: 'http://localhost:3000/ghgas_pipesectionhpa'
            },
            'source-layer': 'ghgas_pipesectionhpa',
            paint: {
                'line-opacity': 1,
                'line-color': 'rgb(53, 175, 109)',
                'line-width': 2
            }
        });
    }
</script>
</body>
</html>
```

**Comment by nyurik** [comment]
I don't know for sure - the way you look the mapbox lib seems ok (i'm not an expert), esp since you use v2 that i have never used (maplibre was forked from v1). I think this is not a Martin issue. Martin generates tiles as MVT vectors for each zoom level. As you zoom in/out, you will get a different tile with different data. Bluriness is not part of it unless mapbox did something weird(?).

**Comment by sharkAndshark** [comment]
Hi @szygiser ,Would you mind to share your data here?  Or would you like to share your SQL if it's a function source?

**Comment by szygiser** [comment]
this is the data






zy.S
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "maplibre/martin"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月8日(星期五) 中午1:25
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [maplibre/martin] Mapboxgl load Martin generated vector slices, small-scale automatic simplification, how to solve? (Issue #1046)





 
Hi @szygiser ,Would you mind to share your data here?

**Comment by szygiser** [comment]
I use maplibre to load, but this issue also occurs

**Comment by nyurik** [comment]
@szygiser please use github web interface. Your email do not get through correctly (i fixed one of your emails). The data did not come in.

**Comment by szygiser** [comment]
> @szygiser please use github web interface. Your email do not get through correctly (i fixed one of your emails). The data did not come in.
[hpa.gdb.zip](https://github.com/maplibre/martin/files/13612550/hpa.gdb.zip)

**Comment by szygiser** [comment]
> Hi @szygiser ,Would you mind to share your data here? Or would you like to share your SQL if it's a function source?

[hpa.gdb.zip](https://github.com/maplibre/martin/files/13612557/hpa.gdb.zip)

**Comment by sharkAndshark** [comment]
@szygiser What's the Spatial Reference here? There's no spatial reference information in your gdb file.

**Comment by szygiser** [comment]
the Spatial Reference&nbsp; is&nbsp;&nbsp;CGCS2000_3_Degree_GK_CM_117E

WKID: 4548&nbsp;&nbsp;



you can&nbsp;convert it to WGS84



zy.S
***@***.***







                       
Original Email
                       
                     

Sender:"Lucas"< ***@***.*** &gt;;

Sent Time:2023/12/13 23:55

To:"maplibre/martin"< ***@***.*** &gt;;

Cc recipient:"szygiser"< ***@***.*** &gt;;"Mention"< ***@***.*** &gt;;

Subject:Re: [maplibre/martin] Mapboxgl load Martin generated vector slices,small-scale automatic simplification, how to solve? (Issue #1046)




 
@szygiser What's the Spatial Reference here? There's no spatial reference information in your gdb file.

**Comment by szygiser** [comment]
> @szygiser What's the Spatial Reference here? There's no spatial reference information in your gdb file.
you can also use the shp file. the Spatial Reference of this file is WGS84.
[shp.zip](https://github.com/maplibre/martin/files/13670255/shp.zip)

**Comment by sharkAndshark** [comment]
I duplicate it with our `debug.thml` @nyurik  Just import the shp.zip and run martin then move to `/8/211/99`

![image](https://github.com/maplibre/martin/assets/42399654/2c6ece8e-f391-43ba-b672-0aa27bf20e91)

**Comment by sharkAndshark** [comment]
And I saved the `7/105/49`(need to remove the `.txt` )

[z7x105y49.mvt.txt](https://github.com/maplibre/martin/files/13675368/z7x105y49.mvt.txt)

**Comment by sharkAndshark** [comment]
That's because they are all too small so [postgis droped them](https://github.com/postgis/postgis/blob/af3e63085419434590f2d52963ca89105e52492f/postgis/lwgeom_out_mvt.c#L86-L109)  @szygiser 
If you merged all lines into one single, it looks normal: 
![image](https://github.com/maplibre/martin/assets/42399654/cc6a187f-30e5-4f4f-adc4-ec8fa9291a3a)
